src/ln/channelmonitor.rs

```diff
@@ -1903,6 +1919,13 @@ impl ChannelMonitor {
 								// has timed out, or we screwed up. In any case, we should now
 								// resolve the source HTLC with the original sender.
 								payment_data = Some(((*source).clone(), htlc_output.payment_hash));
+							} else if !$local_tx {
+								log_claim!($tx_info, $local_tx, htlc_output, false);
+								if let Storage::Local { ref current_remote_commitment_txid, .. } = self.key_storage {
+									check_htlc_valid_remote!(current_remote_commitment_txid, htlc_output);
+								} else if let Storage::Local { ref prev_remote_commitment_txid, .. } = self.key_storage {
+									check_htlc_valid_remote!(prev_remote_commitment_txid, htlc_output);
```

Comment:
This if block can never be hit (and should only run if the previous block failed to set payment_data).

Reply:
Yes, in fact even fail to build a test to this, but wasn't 100% sure it can't be hit